documentation/projects/proposals/document_all_media_properties/20240313-implementation_plan_document_media_properties_api.md

```diff
@@ -0,0 +1,210 @@
+
+    Properties
+    ==========
+    title: >-
+        the title of the media; This documentation can span multiple lines and
+        _can contain Markdown_ as it will be injected as-is into the doc page.
+    """
+
+    pass
+```
+
+### Generate documentation page
+
+With all information in-memory about the media properties, the output will be
+written to a page by the management command. This requires some restructuring in
+the pages created for the catalog media properties documentation.
+
+Similar to the catalog documentation, the output document will include the
+`preamble.md` file from `api/api/docs/media_properties/preamble.md`. This will
+be followed by the Markdown generated from the media properties as a table.
+
+### Verify freshness using CI
+
+Using the existing precedent for API recipes being used in the API tests in the
+CI + CD workflow
+([`django-checks`](https://github.com/WordPress/openverse/blob/68ffb8d4a3b4a2336d25a08972285678f446ec90/.github/workflows/ci_cd.yml#L436)),
+we will invoke `gen-media-props` and check for a zero return status.
+
+## Workstreams
+
+The entire plan can be implemented in one stream. A lot of components of the
+plan are shared with the catalog media properties documentation but code reuse
+between the two projects will be limited as they follow different idioms.
+
+The project assumes that the implementation of the catalog documentation project
+will be completed before this project begins, and includes some reorganisation
+of the output of that project. If that is not the case, the reorganisation can
+be incorporated into the ongoing work on that project.
+
+## Dependencies
+
+We will be using Django's own introspection capabilities. No additional
+libraries other than the already installed Django packages will be needed.
+
+The documentation generated will be added as a page to documentation site
+without needed any new packages.
+
+## Alternatives
+
+The approach for the catalog uses SQL files. Unlike that, we will be using
+Django's introspection capabilities are quite powerful and allow for document
+generation from the models themselves. With Django's migration system, we can
+trust that the output of the introspection will be synced to the state of the
+database.
+
+The approach for the catalog uses a `media_props.md` file for additional notes.
+Unlike that we will be documenting notes inside the model docstrings because it
+allows for the documentation to be close to the code and for the documentation
+to be updated alongside the code.
+
+Although we have used utility scripts for the catalog documentation, for the API
+we will be done using a management command as that is more idiomatic and gives
+us access to the Django ORM models.
+
+## Future improvements
+
+This implementation plan does not document the schema for Django's `JSONField`
+(`jsonb`) in much detail. Further iterations can implement this an as a
+potential improvement.
```

Comment:
Sounds reasonable to me 👍

Reply:
For now, it is possible to document some shape of the file in the additional notes section but we can automate/formalise it in the future using some introspection within Python or from the database.